Simp vax

jonathanhhb opened the pull request:
Added vaccine functions that use SimpleVaccine not TyphoidVaccine. Also added new HINTy example that sets up Rural and Urban HINT groups. Relies on "new old" model branch.

Commit:
Adding example of giving out typhoid vaccines to 9 month olds.

Commit:
Working snapshot using SimpleVaccine for main vax.

Commit:
Use triggered_campaign_delay_event instead of TriggeredCampaignEvent internally for routine_immunization function.

Commit:
Publish VaccineDistributed to RER.

Commit:
Hook up vax mode param.

Commit:
Improved new doc strings.

Commit:
Adding working example adding some HINT groups. Property Value Changing being weird though. Also Property Report.

Commit:
Don't call migrate function. Add requirements.txt that works for this example.

Commit:
Don't call migrate function. Add requirements.txt that works for this example.

shchen-idmod commented on examples/HINTy/manifest.py:
```diff
@@ -0,0 +1,9 @@
+model_dl_dir = "stash2"
+schema_file=os.path.join(model_dl_dir, "schema.json")
+eradication_path=os.path.join(model_dl_dir, "Eradication")
+assets_input_dir="Assets"
```
Can you check in Assets file under this directory?

shchen-idmod commented on examples/vaccination/dtk_centos_2018.id:
Can we put this file under examples directory so we can share it within examples in different folders?

Commit:
New example was missing a couple files.

shchen-idmod committed:
Adding unittests and other type of tests (#12)

"In this series of commits, various enhancements and updates were made to the project. Notably, unittests and workflow tests were added for demographics, alongside the addition of intervention unittests and campaign updates. Several validation improvements were made to demographics unittests, including the introduction of assetcollection IDs for SIF files in 'comps' and 'comps2'. Workflow tests for typhoid vaccine interventions were included, while 'test_demographics_equilibrium_vd_from_worldbank.py' was removed. The Python vaccine tests were expanded and various adjustments were made, including fixing the cutoff day for starting vaccinations, updating vaccine modes, and updating requirements. Further validations were added to 'typhoid_vaccine.py', and coverage sweeps were implemented in vaccine tests. Various code cleanup activities were conducted, such as formatting, imports, and the addition of HINT test cases. Additionally, a login script was added to 'comps', IDm test dependency was removed, and 'outbreak.py' was introduced for interventions. The 'emod-api' version was updated, and additional features like sorting downloaded reporters, GitHub action test scripts, Jenkinsfile updates, and index_url inclusion were implemented." -- ChatGPT

shchen-idmod reviewed (approved):
Passed all tests in Jenkins
